dev/tools/gen_localizations.dart

```diff
@@ -107,19 +107,74 @@ String generateString(String s) {
   return output.toString();
 }
 
+/// Simple data class to hold parsed locale. Does not promise validity of any data.
+class Locale {
+  Locale({
+    this.languageCode,
+    this.scriptCode,
+    this.countryCode,
+    this.length,
+    this.origString
+  });
+
+  String languageCode;
+  String scriptCode;
+  String countryCode;
+  int length;          // The number of fields. Ranges from 1-3.
+  String origString;   // Original un-parsed locale string.
+}
+
+/// Simple parser. Expects the locale string to be in the form of 'language_script_COUNTRY'
+/// where the langauge is 2 characters, script is 4 characters with the first uppercase,
+/// and country is 2-3 characters and all uppercase.
```

Comment:
This all seems reasonable, since we're only dealing with the locales that Flutter supports. 

https://docs.flutter.io/flutter/intl/Intl/canonicalizedLocale.html seems a little more flexible (and also more limited) than this.